Resolved Issue#2 - Added a scoring system

pranavbhasin96 opened the pull request:
I added a scoring system where on correct answer you get 10 points and on wrong answer you get -5 points.
Now, As you can access to a new question only when you have completed all other questions before it, you get +10 score or -5 score only on the respective lastQuestionAllowed. 
Also, i have not put any limit on -5, as many times as the user will give a wrong answer, he will get a -5 (only on the lastQuestionAllowed, ofcourse).

pranavbhasin96 committed:
Added Scoring System

anubhabsen commented:
The check failed, so look into your terminal for errors (server running logs) and fix them.

anubhabsen commented on src/question.js:
```diff
@@ -20,12 +20,21 @@ router.post('/check/:qno(\\d+)?', middleware.isAuthenticated, (req, res) => {
-        if (question.answer == answer && qno ==lastQuestionAllowed) req.user.update({lastQuestionAllowed: lastQuestionAllowed + 1});
+        if (question.answer == answer && qno ==lastQuestionAllowed)
+        {
+	  req.user.update({ score: score + 10});
```
Wouldn't be better if a variable scoreIncrementor be added to score to score instead of 10 so that the scoring system be changed later easily? @meghprkh your thoughts? Also similarly a scoreDecrementor for incorrect responses whose value is 0 by default.

meghprkh replied:
IMO we can put these constants (`scoreIncrement` and `scoreDecrement`) in `config.js` and use them from there

meghprkh commented:
You have messed up the indentation and therefore the CI is not passing. Please install [editorconfig](http://editorconfig.org/) plugin for your editor so that indentation is properly configured. Also if there is an [ESLint](http://eslint.org/) plugin for your editor, please install that or at least run `npm run lint` to check for simple errors.

meghprkh commented on src/question.js:
```diff
@@ -20,12 +20,21 @@ router.post('/check/:qno(\\d+)?', middleware.isAuthenticated, (req, res) => {
+          req.user.update({ lastQuestionAllowed: lastQuestionAllowed + 1 });
+        }
+        else if (question.answer != answer && qno == lastQuestionAllowed)
+        {
```
While not strictly needed I prefer to adhere to the JS standard or semi-standard style http://standardjs.com/ . At least I would like if you put the braces on the same line as `if` and `else if`.

meghprkh replied:
One more style thing, if you have only one line but if is big (in length), then donot use braces, like

``` js
if (x)
  yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy;
else
  zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz;
```

meghprkh commented on src/question.js:
```diff
@@ -20,12 +20,21 @@ router.post('/check/:qno(\\d+)?', middleware.isAuthenticated, (req, res) => {
+        if (question.answer == answer && qno ==lastQuestionAllowed)
+        {
+	  req.user.update({ score: score + 10});
+          req.user.update({ lastQuestionAllowed: lastQuestionAllowed + 1 });
```
Instead of having multiple updates, just have one update query, as in `update({ x : 2, y: 5})`

pranavbhasin96 commented:
In a module like config.js, can i simply add different variables using comma? syntax will be:

`module.exports = {
x: 1,
y: 2,
z: 3
}`

meghprkh commented:
Yes. And then in the other file `const config = require('./config')` and then `config.scoreIncrement`

pranavbhasin96 commented:
const config or var config will do?
you have used var config in auth.js to use the secret.

meghprkh commented:
BTW when you write code enclose in "```" See https://guides.github.com/features/mastering-markdown/

pranavbhasin96 commented:
Its not showing any indentation or syntax error on running `npm run lint`, so its good right?

meghprkh commented:
Yes

pranavbhasin96 committed:
Solved the following problems corresponding to former PR

> Solved indentation problem
> Added flexibility by adding ScoreIncrementor and ScoreDecrementor in config.js
> Issue #2
> PR #10

meghprkh commented:
Thank you and keep the PRs coming =)
